client/src/components/table/index.js

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import './style.css'
+class Table extends React.Component{
+  constructor(props){
+    super(props);
+    this.state={  
+      flightNo :'',
+      flightDate:'',
+      flightTime:'',
+      flightDuration:'',
+      flightsArray:[]
+    }
+    this.handleChange=this.handleChange.bind(this);
+    this.flightDate=this.flightDate.bind(this);
+    this.flightTime=this.flightTime.bind(this);
+    this.flightDuration=this.flightDuration.bind(this);
+  } 
+  
+  handleChange=(e)=>{
+    this.setState({flightNo:e.target.value})
+    this.flightDate();
+    this.flightTime();
+    this.flightDuration();
+  }
+
+  
+  flightDate=(e)=>{
+    this.setState({flightDate:e.target.value})
+  }
+  flightTime=(e)=>{
+    this.setState({flightTime:e.target.value})
+  }
+
+  flightDuration=(e)=>{
+    this.setState({flightDuration:e.target.value})
+  }
+    render(){
+  
+
+      return(
+        
+
+      <div className ="rr">
+          <table id ="table"   onChange={(event)=> this.handleChange(event)}>
```

Comment:
This change handler won't ever get called because tables don't have change events - users can't change them.  I think you won't need this, or the handlers above (flightDate, flightTime, flightDuration)

Once you start displaying real flight data, either from state or from the redux store (so this.props), the render function will handle displaying updates / changes to flight data automatically for you.

Reply:
Plus even if your `handleChange` was actually called the way you think it is going to work @razan99, it still wouldn't work. as you've designed your functions like `flightDate` to receive an `event` argument, the argument you didn't pass in lines 23 through 25.